client/components/PrivateRoute.jsx

```diff
@@ -0,0 +1,21 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { useAuth } from '../store/users/AuthContext';
+
+// https://github.com/WebDevSimplified/React-Firebase-Auth
+export function PrivateRoute({ component: Component, ...rest }) {
+  const { currentUser } = useAuth();
+
+  return (
+    <Route
+      {...rest}
+      render={(props) => (currentUser ? <Component {...props} /> : <Redirect to="/login" />)}
```

Comment:
We should also pass the route so that there's a good redirect after login

Reply:
instead of `{ component: Component, ...rest }`?

Reply:
On the `<Redirect />` we can pass it something - maybe a new route in the URL, so that after login the user is back where they wanted to be